[Kho Tze Jit] iP

tzejit opened the pull request:
## Duke - Your friendly scheduler
> "Please give me A" - Tze Jit

Features include:
* **3** different types of tasks 
* **Filtering** tasks by type *and* date
* **Removing** completed tasks
* **Searching** for tasks

To use Duke simply follow the instructions below:
1. Download it from [here](https://github.com/tzejit/ip/releases/tag/v0.1)
2. Run it in your Command Line with `java -jar filepath\IP.jar`
3. Add tasks and enjoy :innocent::white_check_mark::white_check_mark:

Duke progress:
- [x] Add, remove, and mark tasks as done
- [x] Set deadlines and start times for tasks
- [x] Search for tasks by name and time
- [ ] Implement GUI
- [ ] More exciting features!

Here's a little snippet of the `main` method
```java
public static void main(String[] args) {
        new Duke("duke.txt").run();
    }
```

Commit:
Add Gradle support

tzejit committed:
Add Level-1 functionality

tzejit committed:
Add List class

tzejit committed:
Add Level-2 functionality

tzejit committed:
Add Task class

tzejit committed:
List class: update to contain Tasks

tzejit committed:
Add Level-3 functionality

tzejit committed:
Add Task subclasses

tzejit committed:
Task class: add toString method

tzejit committed:
List class: add toString method

tzejit committed:
Add Level-4 functionality

tzejit committed:
Update Automated Text UI Testing files

tzejit committed:
Add DukeException exception class

tzejit committed:
Add DukeException throwing

tzejit committed:
Add Level-5 functionalities

tzejit committed:
Add Level-6 functionalities

tzejit committed:
Add index exception in DukeException

tzejit committed:
Add function to format String to output

tzejit committed:
Update test case

tzejit committed:
Update .gitignore

tzejit committed:
Add code attribute

tzejit committed:
Add enums to exception types

tzejit committed:
Add Level-7 functionality

tzejit committed:
Add Level-8 functionalities

tzejit committed:
Merge branch 'branch-Level-8'

tzejit committed:
Merge branch 'branch-Level-7'

tzejit committed:
Add A-MoreOOP functionality

tzejit committed:
Update test cases

tzejit committed:
Organize files into packages

tzejit committed:
Update .bat file to accept new packages

tzejit committed:
Add JUnit tests

tzejit committed:
Add help command

tzejit committed:
Create jar file

tzejit committed:
Add JavaDoc comments

tzejit committed:
Modify code to meet coding standards

tzejit committed:
Add Level-9 functionality

tzejit committed:
Merge branch 'branch-A-CodingStandard'

tzejit committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Command.java
#	src/main/java/duke/data/Calendar.java
#	src/main/java/duke/io/Ui.java

tzejit committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Command.java

tzejit committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tzejit committed:
Add gradle

tzejit committed:
Add CheckStyle functionality

tzejit committed:
Fix coding standard violations

tzejit committed:
Add GUI

tzejit committed:
Merge branch 'branch-Level-10'

somekoreanboi commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,235 @@
+                break;
+            default:
+                break;
+            }
```
How about reading the coding standard on switch-cases once again, to make sure this part is correct?

somekoreanboi replied:
Sorry my bad, as we already have confirmed on telegram, your coding standard on switch case is correct and I made a mistake due to a bug in my IDE while opening your file! Sorry for the confusion!

somekoreanboi commented on src/main/java/duke/io/Parser.java:
```diff
@@ -0,0 +1,56 @@
+            return new Command(Command.Commands.ALL, LocalDateTime.parse(aux));
+        default:
+            throw new DukeException(DukeException.Type.COMMAND);
+        }
```
Same as your TaskList class. I like the fact that you are using switch-case instead of if-else, but again, how about reading coding standard on switch-case once again?

somekoreanboi replied:
Same as above; The coding style here is correct and it's not a bug. Sorry!

somekoreanboi commented on src/main/java/duke/io/Ui.java:
```diff
@@ -0,0 +1,171 @@
+                    "  " + list.returnLastTask(), list.returnItemCount(0));
+        default:
+            return "Oops an error has occurred";
+        }
```
Same. How about reconsidering the coding standard for switch-case?

somekoreanboi replied:
Same as above. It's correct!

somekoreanboi commented on src/main/java/duke/io/Ui.java:
```diff
@@ -0,0 +1,171 @@
+            return format(list.getAllBy(dt));
+        default:
+            return "Oops an error has occurred";
+        }
```
Sorry for dupulicates, but how about reconsidering the coding standard for switch-case?

somekoreanboi replied:
Same as above. It's correct!

somekoreanboi commented on src/main/java/duke/io/Ui.java:
```diff
@@ -0,0 +1,171 @@
+            return format("Noted. I've removed this task:", t.toString(), list.returnItemCount(0));
+        default:
+            return "Oops an error has occurred";
+        }
```
Same as above!

somekoreanboi replied:
Same as above. It's correct!

somekoreanboi commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,154 @@
+            return ui.displayCommand(this.command, tasks, this.dateTime);
+        default:
+            return "Oops there is an error";
+        }
```
How about reconsidering your switch-case coding standard?

somekoreanboi replied:
Same as above. It's correct!

somekoreanboi commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,49 @@
+        default:
+            break;
+        }
+    }
```
Sorry for being so same haha, but how about thinking once again on your switch-case indentation!

somekoreanboi replied:
Same as above. It's correct!

somekoreanboi reviewed (commented):
In terms of coding standard violations, very well done!
However, I have noticed you made only one type of critical error, repetitively, which is about switch-case indentation.
Other than that, all perfect! Good job!

somekoreanboi reviewed (commented):
Thank you so much for letting me know that I made a mistake with giving a feedback on your switch-case coding standard Tze Jit! I made a reply to all my comments to make sure it was my mistake. Overall, maybe because you used the checkstyle script to self-evaluate your codes, if we don't count my stupid mistakes, I don't detect any coding standard violations in your code! Very good job!

tzejit committed:
Add bye command

Users can only exit the app by clicking on the cross button.

It is not efficient for users to do so since the app only requires the
use of keyboard.

Let's add a bye command that will allow users to exit the app without
having to switch to a mouse.

It is more efficient for users to continue using a keyboard to exit
instead of switching to a mouse.

tzejit committed:
Add assertions

There are several methods that work based on certain assumptions which
are thought to be checked by their calling methods.

The programmer might leave out certain cases during the checking
process, leading to bugs when the method is called.

Let's add assertions to ensure that these critical assumptions hold true
at the point of method call.

Using assertions help the programmer to identify where mistakes might be
made, and also terminates the app if such mistakes are made to prevent
buggy method calls.

tzejit committed:
Improve code quality by refactoring

tzejit committed:
Merge pull request #2 from tzejit/branch-A-CodeQuality

Improve code quality by refactoring

tzejit committed:
Merge branch 'master' of https://github.com/tzejit/ip into branch-A-CodeQuality

* 'master' of https://github.com/tzejit/ip:

tzejit committed:
Merge branch 'master' into branch-A-Assertions

tzejit committed:
Merge pull request #1 from tzejit/branch-A-Assertions

Add assertions

tzejit committed:
Finalize features

tzejit committed:
Update user guide

tzejit committed:
Set theme jekyll-theme-cayman

tzejit committed:
Add line breaks between commands

tzejit committed:
Merge branch 'master' of https://github.com/tzejit/ip

tzejit committed:
Rename project from duke to chadbot

tzejit committed:
Update user guide

tzejit committed:
Update README.md
